src/ematch_compiler.jl

```diff
@@ -1,34 +1,323 @@
+"""
+We compile a pattern to a backtracking register machine, just like in
+[egg](https://egraphs-good.github.io/), but implemented as a generated Julia
+function rather than a virtual machine.
+
+The "instructions" for this register machine are branches in a large if-else
+statement that looks like:
+
+```julia
+if pc === 1
+  # do instruction 1
+else if pc === 2
+  # do instruction 2
+else
+...
+end
```

Comment:
Let's also explain the label `@label compute` and `@label backtrack` and the gotos